Examples/Java/Sources/Board.java

```diff
@@ -21,42 +20,289 @@
-    public abstract @SerializedName("name") @Nullable String name();
-    public abstract @SerializedName("url") @Nullable String url();
-    public static Builder builder() {
-        return new AutoValue_Board.Builder();
-    }
-    abstract Builder toBuilder();
-    public static TypeAdapter<Board> jsonAdapter(Gson gson) {
-        return new AutoValue_Board.GsonTypeAdapter(gson);
-    }
-    @AutoValue.Builder
-    public abstract static class Builder {
-    
+    @SerializedName("id") private @Nullable String identifier;
+    @SerializedName("contributors") private @Nullable Set<User> contributors;
+    @SerializedName("counts") private @Nullable Map<String, Integer> counts;
+    @SerializedName("created_at") private @Nullable Date createdAt;
+    @SerializedName("creator") private @Nullable Map<String, String> creator;
+    @SerializedName("description") private @Nullable String descriptionText;
+    @SerializedName("image") private @NonNull Image image;
+    @SerializedName("name") private @Nullable String name;
+    @SerializedName("url") private @Nullable String url;
+     static final private int ID_SET = 1 << 0;
+     static final private int CONTRIBUTORS_SET = 1 << 1;
+     static final private int COUNTS_SET = 1 << 2;
+     static final private int CREATED_AT_SET = 1 << 3;
+     static final private int CREATOR_SET = 1 << 4;
+     static final private int DESCRIPTION_SET = 1 << 5;
+     static final private int IMAGE_SET = 1 << 6;
+     static final private int NAME_SET = 1 << 7;
+     static final private int URL_SET = 1 << 8;
+     private int _bits = 0;
+    private Board(@Nullable String identifier,
+    @Nullable Set<User> contributors,
+    @Nullable Map<String, Integer> counts,
+    @Nullable Date createdAt,
+    @Nullable Map<String, String> creator,
+    @Nullable String descriptionText,
+    @NonNull Image image,
+    @Nullable String name,
+    @Nullable String url,
+    int _bits) {
+        this.identifier = identifier;
+        this.contributors = contributors;
+        this.counts = counts;
+        this.createdAt = createdAt;
+        this.creator = creator;
+        this.descriptionText = descriptionText;
+        this.image = image;
+        this.name = name;
+        this.url = url;
+        this._bits = _bits;
+    }
+    public static Board.Builder builder() {
+        return new Board.Builder();
+    }
+    public Board.Builder toBuilder() {
+        return new Board.Builder(this);
+    }
+    public Board mergeFrom(Board model) {
+        Board.Builder builder = this.toBuilder();
+        builder.mergeFrom(model);
+        return builder.build();
+    }
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        Board that = (Board) o;
+        return Objects.equals(this.identifier, that.identifier) &&
```

Comment:
For later, in ObjC we sorted the equality comparisons to compare primitive properties first since that was a bit faster

Reply:
I'll file a ticket for this as well